src/ui/win32/DialogBase.cpp

```diff
@@ -60,25 +68,26 @@ INT_PTR CALLBACK DialogBase::DialogProc(_UNUSED HWND hDlg, UINT uMsg, WPARAM wPa
             return 0;
 
         case WM_COMMAND:
-            return OnCommand(LOWORD(wParam));
+        {
+            const auto id{ ra::to_signed(LOWORD(wParam)) };
+            if (id < 1)
+                return -1; // Invalid Command ID
+            OnCommand(id);
+        }break;
```

Comment:
This is how the API handles/forwards WM_COMMAND by default.
```
/* void Cls_OnCommand(HWND hwnd, int id, HWND hwndCtl, UINT codeNotify) */
#define HANDLE_WM_COMMAND(hwnd, wParam, lParam, fn) \
    ((fn)((hwnd), (int)(LOWORD(wParam)), (HWND)(lParam), (UINT)HIWORD(wParam)), 0L)
#define FORWARD_WM_COMMAND(hwnd, id, hwndCtl, codeNotify, fn) \
    (void)(fn)((hwnd), WM_COMMAND, MAKEWPARAM((UINT)(id),(UINT)(codeNotify)), (LPARAM)(HWND)(hwndCtl))
```
break after the closing brace seems good to me.

Reply:
So instead of writing the block explicit there you could just write.
`HANDLE_MSG(hwnd, WM_COMMAND, this->OnCommand);`
It won't work with current one but that's OK I guess.